Source/Core/Core/CheatSearch.cpp

```diff
@@ -572,11 +572,20 @@ std::string Cheats::CheatSearchSession<T>::GetResultValueAsString(size_t index,
   if (hex)
   {
     if constexpr (std::is_same_v<T, float>)
-      return fmt::format("0x{0:08x}", Common::BitCast<u32>(m_search_results[index].m_value));
+    {
+      return fmt::format("0x{0:08x}", Common::BitCast<s32>(m_search_results[index].m_value));
+    }
     else if constexpr (std::is_same_v<T, double>)
-      return fmt::format("0x{0:016x}", Common::BitCast<u64>(m_search_results[index].m_value));
+    {
+      return fmt::format("0x{0:016x}", Common::BitCast<s64>(m_search_results[index].m_value));
+    }
     else
-      return fmt::format("0x{0:0{1}x}", m_search_results[index].m_value, sizeof(T) * 2);
+    {
+      return fmt::format(
+          "0x{0:0{1}x}",
+          Common::BitCast<typename std::make_unsigned<T>::type>(m_search_results[index].m_value),
```

Comment:
Can't `std::make_unsigned_t` be used here?

Reply:
Indeed it can.